### PR TITLE
Do not transcode H264 video track egress

### PR DESCRIPTION
**Pull request by biglittlebigben**

### CODE REVIEW

File: pkg/pipeline/input/builder/video.go [hunk 1]
```diff
@@ -107,7 +107,7 @@ func (v *VideoInput) buildSDKDecoder(p *config.PipelineConfig, src *app.Source,
 	case strings.EqualFold(codec.MimeType, string(types.MimeTypeH264)):
 		if err := src.Element.SetProperty("caps", gst.NewCapsFromString(
 			fmt.Sprintf(
-				"application/x-rtp,media=video,payload=%d,encoding-name=H264,clock-rate=%d",
+				"application/x-rtp,media=video,payload=%d,encoding-name=H264,clock-rate=%d,width=1980,height=1080",
```
Comment (frostbyte73) [inline pkg/pipeline/input/builder/video.go:110]: p.Width/Height?

**Review by frostbyte73 — APPROVED** [review]

File: pkg/pipeline/input/builder/video.go (hunk 1 above)
Comment (davidzhao) [reply pkg/pipeline/input/builder/video.go:110]: FYI: it's possible for TrackInfo.width to be 0 sometimes. It's a rare condition. we should have some defaults for this if possible.

Also, do the values set here do anything? or is it just to get the downstream caps to be happy?

Comment (biglittlebigben) [reply pkg/pipeline/input/builder/video.go:110]: Thanks for catching this. This is test code that made its way in the PR. the h264parse step will extract the actual dimensions.

**Review by davidzhao — APPROVED** [review]
lgtm

**Comment by biglittlebigben** [comment]
Track composite allows you to force transcoding. This will however not do framerate conversion unless the source framerate is above the encoder framerate setting. You'll need to use RoomComposite to reach a constant framerate. Framerate conversion will however inherently create a somewhat jittery output (which is why we try to avoid it when possible)

**Pull request merged**